db/change_cache.go

```diff
@@ -277,8 +277,9 @@ func (c *changeCache) InsertPendingEntries(ctx context.Context) error {
 	// Trigger _addPendingLogs to process any entries that have been pending too long:
 	c.lock.Lock()
 	changedChannels := c._addPendingLogs(ctx)
-	if c.notifyChange != nil && len(changedChannels) > 0 {
-		c.notifyChange(ctx, changedChannels)
+	channelSet := channels.SetFromArrayNoValidate(changedChannels)
+	if c.notifyChange != nil && len(channelSet) > 0 {
+		c.notifyChange(ctx, channelSet)
 	}
```

Comment:
`channels.SetFromArrayNoValidate` is very cheap to run (one empty allocation) if `changedChannels` is empty, but there also no reason to run the function conceptually.

```suggestion
	if c.notifyChange != nil && len(changedChannels) > 0 {
		channelSet := channels.SetFromArrayNoValidate(changedChannels)
		c.notifyChange(ctx, channelSet)
}
```